### ISSUE TITLE
Replace docker registry using regex manager

### ISSUE THREAD

**Issue by jsok** [renovatebot/renovate#6938]
### Which Renovate are you using?

Renovate Open Source CLI

### Which platform are you using?

GitHub Enterprise

### Have you checked the logs? Don't forget to include them if relevant

Yes

### What would you like to do?

I am currently undertaking a migration from a legacy Docker registry to a new registry.
I would like to use the regex manager to detect updates and rewrite the registry as well as the version.

e.g. I'd like to migrate `FROM legacy.example.com/image:tag` to `FROM new.example.com/image:tag`

I have a regex config which looks like:

```javascript
  regexManagers: [
    {
      fileMatch: ['(^|/|\\.)Dockerfile$', '(^|/)Dockerfile\\.[^/]*$'],
      matchStrings: [
        "FROM legacy\.example\.com\/(?<repo>.*?):(?<currentValue>.*?)@(?<currentDigest>.*)\n"
      ],
      datasourceTemplate: "docker",
      depNameTemplate: "legacy.example.com/{{repo}}",
      lookupNameTemplate: "new.example.com/{{repo}}",
    }
]
```

which is correctly getting updates from the new registry (I have verified the digests).

However it only updates the tag/digest, and leaves the registry untouched in my PRs.
Logs below indicate this:

```raw
DEBUG: packageFiles with updates (repository=renovate-example)
       "config": {
         "regex": [
           {
             "packageFile": "Dockerfile",
             "manager": "regex",
             "deps": [
               {
                 "depName": "legacy.example.com/library/alpine",
                 "lookupName": "new.example.com/library/alpine",
                 "currentValue": "3.12",
                 "currentDigest": "sha256:1eb241ae6bc3e755609e210543ae3803b0e1ddae68183289b065d89ab42d9e5c",
                 "datasource": "docker",
                 "replaceString": "FROM legacy.example.com/library/alpine:3.12@sha256:1eb241ae6bc3e755609e210543ae3803b0e1ddae68183289b065d89ab42d9e5c\n",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": []
               }
             ],
             "matchStrings": [
               "FROM legacy.example.com/(?<repo>.*?):(?<currentValue>.*?)@(?<currentDigest>.*)\n"
             ]
           }
         ]
       }
```

The `replaceString` would ideally be `FROM new.example.com/library/alpine:3.12@sha256:1eb241ae6bc3e755609e210543ae3803b0e1ddae68183289b065d89ab42d9e5c`

Can I make such a change using renovate?

**Comment by refnode** [comment]
Hi,

I also explore atm, how I can extend renovate to use something like a map for replace fqdn's / path patterns and strings / variables a transparent way for a transition on a hugher repo base.

I think it makes sense to have that use case available for renovate.

**Comment by viceice** [comment]
This can maybe be solved with #5558

**Comment by secustor** [comment]
We have now a tutorial which describes how another user has achieved this: https://docs.renovatebot.com/modules/manager/regex/

Feel free to reopen this issue if you disagree.
